rclpy/test/test_lifecycle.py

```diff
@@ -63,7 +69,19 @@ def test_lifecycle_state_transitions():
     with pytest.raises(_rclpy.RCLError):
         node.trigger_deactivate()
     assert node.trigger_shutdown() == TransitionCallbackReturn.SUCCESS
+    with pytest.raises(_rclpy.RCLError):
+        node.trigger_shutdown()
     node.destroy_node()
+    # Again but trigger shutdown from 'inactive' instead of 'unconfigured'
+    node = LifecycleNode(
+        'test_lifecycle_state_transitions_2', enable_communication_interface=False)
+    assert node.trigger_shutdown() == TransitionCallbackReturn.SUCCESS
+    # Again but trigger shutdown from 'active'
+    node = LifecycleNode(
+        'test_lifecycle_state_transitions_3', enable_communication_interface=False)
+    assert node.trigger_configure() == TransitionCallbackReturn.SUCCESS
+    assert node.trigger_activate() == TransitionCallbackReturn.SUCCESS
```

Comment:
Maybe something like the following to show that transitions work robustly? You can also do this kind of thing elsewhere where there is a possible loop.

```suggestion
    for _ in range(10):
      assert node.trigger_configure() == TransitionCallbackReturn.SUCCESS
      assert node.trigger_cleanup() == TransitionCallbackReturn.SUCCESS
    assert node.trigger_configure() == TransitionCallbackReturn.SUCCESS
    for _ in range(10):
      assert node.trigger_activate() == TransitionCallbackReturn.SUCCESS
      assert node.trigger_deactivate() == TransitionCallbackReturn.SUCCESS
    assert node.trigger_activate() == TransitionCallbackReturn.SUCCESS
```

Reply:
done in https://github.com/ros2/rclpy/pull/865/commits/c69339833e41dd12a127958e9999d0dba68d3ac8